opentelemetry_api/opentelemetry/status.h

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2019, OpenTelemetry Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#ifndef OPENTELEMETRY_STATUS_H_
+#define OPENTELEMETRY_STATUS_H_
+
+#include <stdint.h>
+
+typedef int32_t status_t;
```

Comment:
If we're going to use error codes, I'd rather use the standard library's [std::error_code](https://en.cppreference.com/w/cpp/error/error_code).

It's somewhat of a pain to work with, but IMO it's still better than rolling your own